src/cdomains/valueDomain.ml

```diff
@@ -115,7 +115,7 @@ struct
     | _ -> false
 
   let is_mutex_type (t: typ): bool = match t with
-    | TNamed (info, attr) -> info.tname = "pthread_mutex_t" || info.tname = "spinlock_t" || info.tname = "pthread_spinlock_t"
+    | TNamed (info, attr) -> info.tname = "pthread_mutex_t" || info.tname = "spinlock_t" || info.tname = "pthread_spinlock_t" || info.tname = "pthread_cond_t"
```

Comment:
Why are we treating `pthread_cond_t` as a mutex type here? That is confusing.

Reply:
It wouldn't be more consistent though because it still uses `Mutex` for `ValueDomain`. I don't think it's worth it to introduce `Cond` into `ValueDomain` just for this. Mutex analyses treat condition variables via unlock-lock anyway.

Reply:
> Mutex analyses treat condition variables via unlock-lock anyway.

What do you mean by this? If there is an analysis that does this, it is wrong. (Or are you talking about the second argument of `pthread_cond_wait`? That seems to be unrelated, no?

Reply:
And we do something with condition variables, see #520.

Reply:
Well yes, but we don't care about the implementation detail struct of a condition variable in base, just like we don't care about it for mutexes. All this does is make the contents opaque instead of tracking a struct whose contents our `special` functions don't update and thus those contents are outright wrong.